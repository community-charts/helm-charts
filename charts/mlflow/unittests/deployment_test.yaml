suite: test deployment

templates:
  - deployment.yaml

tests:
  - it: should be memory backended when postgres is not enabled
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[3]
          value: --backend-store-uri=sqlite:///:memory

  - it: should be postgres backended when postgres is enabled
    set:
      backendStore.postgres.enabled: true
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[3]
          value: --backend-store-uri=postgresql://

  - it: should be local folder as an artifact root when azureBlob or s3 or gcs is not enabled
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[4]
          value: --default-artifact-root=./mlruns

  - it: should be azureBlob as an artifact root when azureBlob is enabled
    set:
      artifactRoot.azureBlob.enabled: true
      artifactRoot.azureBlob.container: test-container
      artifactRoot.azureBlob.storageAccount: test-account
      artifactRoot.azureBlob.path: mlflow_folder
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[4]
          value: --default-artifact-root=wasbs://test-container@test-account.blob.core.windows.net/mlflow_folder

  - it: should be s3 as an artifact root when s3 is enabled
    set:
      artifactRoot.s3.enabled: true
      artifactRoot.s3.bucket: test-bucket
      artifactRoot.s3.path: mlflow_folder
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[4]
          value: --default-artifact-root=s3://test-bucket/mlflow_folder

  - it: should expose prometheus metrics when serviceMonitor is enabled
    set:
      serviceMonitor.enabled: true
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[5]
          value: --expose-prometheus=/mlflow/metrics

  - it: should show extra arguments when we pass any
    set:
      extraArgs:
        workers: "4"
        staticPrefix: "mlflow"
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[6]
          value: --workers=4
      - equal:
          path: spec.template.spec.containers[0].args[5]
          value: --static-prefix=mlflow

  - it: should show extra flags when we pass any
    set:
      extraFlags:
        - serveArtifacts
        - artifactsOnly
    asserts:
      - equal:
          path: spec.template.spec.containers[0].args[5]
          value: --serve-artifacts
      - equal:
          path: spec.template.spec.containers[0].args[6]
          value: --artifacts-only

  - it: should show extra secret environment variables from secrets when we pass any
    set:
      extraSecretNamesForEnvFrom:
        - mysupersecret
    asserts:
      - equal:
          path: spec.template.spec.containers[0].envFrom[2].secretRef.name
          value: mysupersecret

  - it: should match snapshot of default values
    set:
      podAnnotations:
        test-key: test-value
      podSecurityContext:
        fsGroup: 2000
      securityContext:
        runAsNonRoot: true
        runAsUser: 1000
      initContainers:
        - name: init-myservice
          image: busybox:1.28
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      extraVolumeMounts:
        - name: mlflow-volume
          mountPath: /mlflow/data
        - name: example
          mountPath: /example
      extraVolumes:
        - name: mlflow-volume
          persistentVolumeClaim:
            claimName: mlflow-pv-volume
        - name: example
          configMap:
            name: example
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
