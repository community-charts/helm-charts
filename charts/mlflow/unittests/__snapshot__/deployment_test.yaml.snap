should match snapshot of default values:
  1: |
    apiVersion: v1
    data:
      DO_NOT_TRACK: "true"
      MLFLOW_CONFIGURE_LOGGING: "true"
      MLFLOW_DISABLE_TELEMETRY: "true"
      MLFLOW_LOGGING_LEVEL: INFO
    kind: ConfigMap
    metadata:
      labels:
        app: mlflow
        chart: mlflow-1.0.0
        heritage: Helm
        release: mlflow
      name: mlflow-env-configmap
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: mlflow
        app.kubernetes.io/instance: mlflow
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: mlflow
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: mlflow-1.0.0
        version: 1.0.0
      name: mlflow
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: mlflow
          app.kubernetes.io/instance: mlflow
          app.kubernetes.io/name: mlflow
      strategy:
        rollingUpdate:
          maxSurge: 100%
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          annotations:
            checksum/config: 6a1114d91915da5454596d7c7d7f7ae694d835f570534869b7d474e0638ca692
          labels:
            app: mlflow
            app.kubernetes.io/instance: mlflow
            app.kubernetes.io/name: mlflow
        spec:
          containers:
            - args:
                - server
                - --host=0.0.0.0
                - --port=5000
                - '--backend-store-uri=sqlite:///:memory:'
                - --default-artifact-root=./mlruns
                - --gunicorn-opts='--log-level=info'
              command:
                - mlflow
              env:
                - name: MLFLOW_VERSION
                  value: 1.0.0
              envFrom:
                - configMapRef:
                    name: mlflow-env-configmap
                - secretRef:
                    name: mlflow-env-secret
                - secretRef:
                    name: mlflow-flask-server-secret-key
              image: burakince/mlflow:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health
                  port: http
                initialDelaySeconds: 10
                periodSeconds: 30
                timeoutSeconds: 3
              name: mlflow
              ports:
                - containerPort: 5000
                  name: http
                  protocol: TCP
              readinessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health
                  port: http
                initialDelaySeconds: 10
                periodSeconds: 30
                timeoutSeconds: 3
              resources: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                privileged: false
                readOnlyRootFilesystem: false
                runAsGroup: 1001
                runAsNonRoot: true
                runAsUser: 1001
          securityContext:
            fsGroup: 1001
            fsGroupChangePolicy: OnRootMismatch
          serviceAccountName: mlflow
